src/lib.rs

```diff
@@ -42,4 +81,4 @@ impl Drop for XenEventChannel {
             xenevtchn_sys::xenevtchn_close(self.handle);
```

Comment:
You need to call `xenevtchn_unbind()` here, since you bind in the init function.